feat: [sc-63726] Add auto-completion to the Vendor CLI

FourSigma opened the pull request:
Story details: https://app.shortcut.com/replicated/story/63726

`replicated-cli` PR:  https://github.com/replicatedhq/replicated/pull/313

Usage: 
```
╰─$ ./replicated completion --help


	Bash:

		This script depends on the 'bash-completion' package.
		If it is not installed already, you can install it via your OS's package manager.

	  $ source <(replicated completion bash)

	  # To load completions for each session, execute once:
	  # Linux:
	  $ replicated completion bash > /etc/bash_completion.d/replicated
	  # macOS:
	  $ replicated completion bash > $(brew --prefix)/etc/bash_completion.d/replicated

	Zsh:

	  # If shell completion is not already enabled in your environment,
	  # you will need to enable it.  You can execute the following once:

	  $ echo "autoload -U compinit; compinit" >> ~/.zshrc

	  # To load completions for each session, execute once:
	  $ replicated completion zsh > "${fpath[1]}/_replicated"

	  # You will need to start a new shell for this setup to take effect.

	fish:

	  $ replicated completion fish | source

	  # To load completions for each session, execute once:
	  $ replicated completion fish > ~/.config/fish/completions/replicated.fish

	PowerShell:

	  PS> replicated completion powershell | Out-String | Invoke-Expression

	  # To load completions for every new session, run:
	  PS> replicated completion powershell > replicated.ps1
	  # and source this file from your PowerShell profile.


Usage:
  replicated completion [bash|zsh|fish|powershell]

Flags:
  -h, --help   help for completion

Global Flags:
      --app string     The app slug or app id to use in all calls
      --token string   The API token to use to access your app in the Vendor API

```

@paigecalvert Should this detailed help text be the docs PR?

FourSigma commented:
Here an example from the GitHub CLI - https://cli.github.com/manual/gh_completion

paigecalvert commented:
@FourSigma thank you, looks great! I just added subtopics for all the commands and added everything to the sidebar. And I feel like I am leaning towards leaving the detailed help info out of the docs since we do tell them about the --help flag, and including the help text in the docs as well is probably a good recipe for it becoming out of date quickly. what do you think?

FourSigma commented:
> @FourSigma thank you, looks great! I just added subtopics for all the commands and added everything to the sidebar. And I feel like I am leaning towards leaving the detailed help info out of the docs since we do tell them about the --help flag, and including the help text in the docs as well is probably a good recipe for it becoming out of date quickly. what do you think?

That works for me Paige!
